adrs/4937890790/README.md

```diff
@@ -0,0 +1,151 @@
+# JavaScript Modules exports convention
+
+* **State:** Draft
+* **Created:** 2021-02-17
+* **Tags:** js,ts
+
+## Context
+
+We don't have a consistent way to export a module's functions and variables
+across the JavaScript ecosystem.
+
+Although such concern is not a big concern for more Senior developers, there is
+some cognitive load when it comes to default exports over named exports.
+
+What is one of the most challenging things in computer science? Naming.
+
+Using default-exports over named-exports forces the user to think about the name
+of the import.
+
+Let's use the following example:
+
+```ts
+import [name here] from 'base-button'
+```
+
+What should be the right name for the import?
+
+* `Button`
+* `baseButton`
+* `BaseButton`
+* `MyButton` or any other name you would like it to be
+
+For a simple import, you have already at least three different variations to
+think about it, or combinations of any name for that matter.
+
```

Comment:
I know this comes up later, but I think it's horribly BM to rename imports -- even  in the `export default` world.  😂 


```ts
import Button from 'base-button'
```




In some cases, you have to avoid name conflict.

```ts
import TheirBaseButton from 'base-button'
import OurBaseButton from '../base-button'
```


Besides that, why would anybody do that?